### PR TITLE
Fix whitelisting of non-existing ks/table

### PR DESCRIPTION
**Pull request by eperott**

### CODE REVIEW

File: src/main/java/com/ericsson/bss/cassandra/ecaudit/AuditEntryBuilderFactory.java [hunk 1]
```diff
@@ -62,20 +73,49 @@
...
     public Builder createEntryBuilder(String operation, ClientState state)
     {
-        CQLStatement statement;
         try
         {
-            statement = QueryProcessor.getStatement(operation, state).statement;
+            CQLStatement statement = QueryProcessor.getStatement(operation, state).statement;
+            return createEntryBuilder(statement);
+        }
+        catch (KeyspaceNotDefinedException e)
```
Comment (emolsson) [inline src/main/java/com/ericsson/bss/cassandra/ecaudit/AuditEntryBuilderFactory.java]: This seems to be specific to a keyspace, would we get into a similar scenario if a table does not exist?

File: src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/ITVerifyAudit.java [hunk 2]
```diff
@@ -531,6 +525,45 @@ public void testFailedStatementsAreLogged()
                         .containsAll(expectedAttemptsAndFails(statements));
     }
 
+    @Test
```
Comment (emolsson) [inline src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/ITVerifyAudit.java]: I might be missing it, but I'd like to see a test case for a failed batch statement which is not whitelisted.

Comment (eperott) [reply src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/ITVerifyAudit.java]: Known issue: Ericsson/ecaudit#11

File: src/main/java/com/ericsson/bss/cassandra/ecaudit/AuditEntryBuilderFactory.java [hunk 3]
```diff
@@ -78,7 +79,7 @@ public Builder createEntryBuilder(String operation, ClientState state)
             CQLStatement statement = QueryProcessor.getStatement(operation, state).statement;
             return createEntryBuilder(statement);
         }
-        catch (KeyspaceNotDefinedException e)
+        catch (InvalidRequestException e)
```
Comment (emolsson) [inline src/main/java/com/ericsson/bss/cassandra/ecaudit/AuditEntryBuilderFactory.java]: If we throw an exception from within the catch clause here we would throw it through the stack. Should we have "catch (RunTimeException)" outside of this try/catch?

Comment (eperott) [reply src/main/java/com/ericsson/bss/cassandra/ecaudit/AuditEntryBuilderFactory.java]: Done

**Pull request merged**